src/arraytypes/arraytypes.jl

```diff
@@ -35,14 +35,19 @@ function toarrowvector(x, i=1, de=Dict{Int64, Any}(), ded=DictEncoding[], meta=g
     @debugv 2 "converting top-level column to arrow format: col = $(typeof(x)), compression = $compression, kw = $(kw.data)"
```

Comment:
This function should probably be broken down into a few smaller functions for clarity:

```julia

toarrowvector(a::ArrowVector, compression::Nothing, kw...) = a

function toarrowvector(a::ArrowVector, compression::AbstractVector{LZ4FrameCompressor}, kw...)
    tid=Threads.threadid()
    lock(LZ4_FRAME_COMPRESSOR_LOCK[tid]) do
        compress(Meta.CompressionTypes.LZ4_FRAME, compression[tid], a)
    end
end

function toarrowvector(a::ArrowVector, compression::LZ4FrameCompressor, Vector{ZstdCompressor}, ZstdCompressor}=nothing, kw...)
    compress(Meta.CompressionTypes.LZ4_FRAME, compression, a)
end

function toarrowvector(a::ArrowVector, compression::AbstractVector{ZstdCompressor}, kw...)
    compress(Meta.CompressionTypes.ZSTD, compression, a)
end

function toarrowvector(a::ArrowVector, compression::ZstdCompressor, kw...)
    tid=Threads.threadid()
    lock(ZSTD_COMPRESSOR_LOCK[tid]) do
        compress(Meta.CompressionTypes.ZSTD, compression[tid], a)
    end
end

function toarrowvector(x, i=1, de=Dict{Int64, Any}(), ded=DictEncoding[], meta=getmetadata(x); compression::Union{Nothing, Vector{LZ4FrameCompressor}, LZ4FrameCompressor, Vector{ZstdCompressor}, ZstdCompressor}=nothing, kw...)
    @debugv 2 "converting top-level column to arrow format: col = $(typeof(x)), compression = $compression, kw = $(kw.data)"
    @debugv 3 x
    A = arrowvector(x, i, 0, 0, de, ded, meta; compression=compression, kw...)
    A = toarrowvector(A, compression)
    @debugv 2 "converted top-level column to arrow format: $(typeof(A))"
    @debugv 3 A
    return A
end


```